pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -234,14 +239,56 @@ func (r *MachineSetSyncReconciler) syncMachineSets(ctx context.Context, mapiMach
 
 		return ctrl.Result{}, nil
 	}
-
-	return ctrl.Result{}, nil
 }
 
 // reconcileMAPIMachineSetToCAPIMachineSet reconciles a MAPI MachineSet to a CAPI MachineSet.
 func (r *MachineSetSyncReconciler) reconcileMAPIMachineSetToCAPIMachineSet(ctx context.Context, mapiMachineSet *machinev1beta1.MachineSet, capiMachineSet *capiv1beta1.MachineSet) (ctrl.Result, error) {
-	// This function is currently a placeholder.
-	return ctrl.Result{}, nil
+	logger := log.FromContext(ctx)
+
+	newCAPIMachineSet, newCAPIInfraMachineTemplate, warns, err := r.convertMAPIToCAPIMachineSet(mapiMachineSet, r.Infra)
```

Comment:
Are we setting `r.Infra` anywhere? A quick cmd+f and I can't see where we would be, and this would panic otherwise.

Reply:
Also is there a reason this stores the OCP Infra on the reconciler struct, rather than using a helper func e.g `fetchOCPInfra`? (where the CAPI counterpart is `fetchCAPIInfraResources`? )

It feels like two different ways of doing things, one for CAPI and one for MAPI. Maybe we should standardise to one?

Reply:
Yes we do se it in `cmd/machine-api-migration/main.go` of this PR

```
	machineSyncReconciler := machinesync.MachineSyncReconciler{
		Infra:    infra,
```

And it stores the OCP Infra object in the reconciler because we are already getting it in the main.go file to extract the provider from it, and so we can just plug it in that way and use it in the various controllers that require it.

This object is not expected to be changed while our controller is running. This though is not the case for the CAPI Infra resources, which will change during the controller normal runtime.

Reply:
So considering they are different things and we have different expectations and uses for them, I would keep them like they are now.

Reply:
> Yes we do se it in cmd/machine-api-migration/main.go of this PR

Ah I'd missed that - nice! 

> This object is not expected to be changed while our controller is running. This though is not the case for the CAPI Infra resources, which will change during the controller normal runtime.

Cool - that makes sense to me! :) Cheers for clearing up! 


After chatting on slack we're not too sure what the nil poniter deref the QE team have managed to create. Will need some digging in order to repro.